### ISSUE TITLE
Drop a runtime in futures

### ISSUE THREAD

**Issue by lerencao** [tokio-rs/tokio#1950]
## Version
tokio v0.2.4

## Platform
Darwin Kernel Version 19.0.0

## Description

It's not possible to drop a tokio runtime in async context.

for example:

``` rust
struct A {
    _rt: Runtime,
}

impl A {
    pub async foo(&self) {}
}

struct B {
    a: A,
}

impl B{
   pub async bar(&self) { self.a.foo.await }
}

pub async foobar(b: B) {
    b.bar()
    // here, dropping `a` will fail.  dropping `BlockingPool` needs `enter`, 
    //which is not available in context of futures.
}
```

Fow now, I have to keep track of every usage on structs who contain `Runtime`, and make sure they are dropped in non-async context.

**Comment by carllerche** [comment]
Dropping a runtime is intended to be a blocking operation. 

Can you tell me why you need to hold a runtime from a future? Can you hold a runtime::Handle instead?

**Comment by lerencao** [comment]
@carllerche 

Our code embraces async using actor model. 
And different actors may  run in a different runtime, which is hold by a supervisor actor.  
When stopping the supervisor actor, we also call the stop function using async. 
In this case we encountered the problem.

If dropping a runtime is a blocking operation, I think we may need to change the way of handling runtime.

**Comment by LucioFranco** [comment]
You shouldn't need multiple runtimes to run an actor system, have you taken a look #1879? Might have some better solutions to this problem.

**Comment by lerencao** [comment]
@LucioFranco  thanks for the issue!
Will feedback later on this.

**Comment by Darksonn** [comment]
Fixed by #2646.
